### PR TITLE
Move to gomodules in preparation for kubebuilder v2 migration

### PR DESCRIPTION
**Pull request by maxsmythe**
Signed-off-by: Max Smythe <smythe@google.com>

### CODE REVIEW

File: constraint/Makefile [hunk 1]
```diff
@@ -33,8 +40,18 @@ vet:
...
 
 # Generate code
-generate:
+generate: .output/buildutil
 ifndef GOPATH
 	$(error GOPATH not defined, please define GOPATH. Run "go help gopath" to learn more about GOPATH)
 endif
-	go generate ./pkg/...
+	export GOROOT=$(go env root)
+	GO111MODULE=off go generate ./pkg/...
```
Comment (ritazh) [inline constraint/Makefile]: Getting the following on my mac:

```
export GOROOT=
GO111MODULE=off go generate ./pkg/...
pkg/apis/apis.go:17: running "../../.output/buildutil/deepcopy-gen": fork/exec ../../.output/buildutil/deepcopy-gen: exec format error
pkg/core/templates/constrainttemplate_types.go:17: running "../../../.output/buildutil/deepcopy-gen": fork/exec ../../../.output/buildutil/deepcopy-gen: exec format error
make[1]: *** [generate] Error 1
make: *** [build-constraint] Error 2
```

We can either make sure everything is built/generated in a container or make sure things run natively on mac. WDYT?

File: build/build_constraint.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+
+cd constraint
+make
```
Comment (ritazh) [inline build/build_constraint.sh]: nit: newline

File: constraint/buildutils/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Build the manager binary
+FROM golang:1.13.3 as builder
+
+WORKDIR /scratch
+
+COPY . .
+
+ENTRYPOINT ./maketools.sh
```
Comment (ritazh) [inline constraint/buildutils/Dockerfile]: nit: newline

File: constraint/buildutils/maketools.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+echo 'Starting tool build...'
+
+echo 'controller-gen'
+go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.1.8
```
Comment (sozercan) [inline constraint/buildutils/maketools.sh]: v0.2.2?

File: constraint/Makefile [hunk 5]
```diff
@@ -33,8 +40,18 @@ vet:
...
+
+# find or download controller-gen
+# download controller-gen if necessary
+.output/buildutil:
+	mkdir -p .output
+	docker build buildutils -t maketools && docker run -v ${PWD}/.output:/.output maketools
+	echo "buildutils generated:"
+	ls -l .output/buildutil
```
Comment (ritazh) [inline constraint/Makefile]: this is listed twice here and in maketools

File: constraint/buildutils/maketools.sh [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+echo 'conversion-gen'
+go get k8s.io/code-generator/cmd/conversion-gen@93d7507fc8ffb4c860e4dd6f9a51a5e1985aebcd
+
+mkdir /.output/buildutil
+chmod 755 /.output/buildutil
+cp /go/bin/* /.output/buildutil/.
+
+echo 'Contents of /.output/buildutil are:'
+ls -l /.output/buildutil
```
Comment (ritazh) [inline constraint/buildutils/maketools.sh]: nit: newline

File: constraint/Dockerfile [hunk 7]
```diff
@@ -1,5 +1,5 @@
 # Build the manager binary
-FROM golang:1.10.3 as builder
+FROM golang:1.13.3 as builder
```
Comment (ritazh) [inline constraint/Dockerfile]: https://github.com/open-policy-agent/frameworks/blob/e6de551bda0b30151ef166c1ac9f988b585600c7/constraint/Dockerfile#L9 `2.0.1`?

File: constraint/Makefile (hunk 1 above)
Comment (maxsmythe) [reply constraint/Makefile]: Fixed to run natively

File: constraint/buildutils/maketools.sh (hunk 4 above)
Comment (maxsmythe) [reply constraint/buildutils/maketools.sh]: Per the README in this directory, we have to be sure we are using an earlier version of controller-gen, as the newer version doesn't work with the CRD resource.

File: constraint/Makefile (hunk 5 above)
Comment (maxsmythe) [reply constraint/Makefile]: Moot now (removed).

**Pull request merged**